src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1943,6 +1944,110 @@ internal void IncrementPasteOffset()
             this.currentPasteOffset = (this.currentPasteOffset + PasteOffsetStep) % PasteOffsetMax;
         }
 
+        #region [Nodes Info]
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings
+        /// </summary>
+        internal bool HasWarnings
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Warning || n.State == ElementState.PersistentWarning); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings with no Geometry
+        /// </summary>
+        internal bool HasWarningsWithNoGeometry
+        {
+            get { return Nodes.Any(n => (n.State == ElementState.Warning || n.State == ElementState.PersistentWarning) && !n.Category.StartsWith("Geometry.")); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if workspace run with errors
+        /// </summary>
+        internal bool HasErrors
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Error); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if home workspace is displayed with infos
+        /// </summary>
+        internal bool HasInfos
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Info); }
+        }
+
+        /// <summary>
+        /// All the Nodes identified as deprecated
+        /// </summary>
+        internal IEnumerable<NodeModel> DeprecatedNodes
+        {
+            get
+            {
+                return Nodes.Where(node => node.NodeInfos.Any(nodeInfo => nodeInfo.Message.Contains(Configurations.DeprecatedMethodMessage)));
```

Comment:
Here you're doing a string comparison to check if a node is deprecated or not. This is brittle as the deprecated message can change over time and we would need to keep the `Configurations.DeprecatedMethodMessage` property in sync with that. Instead of this, I would suggest we check for the attribute on the node such as `IsObsolete` or one of those other obsolete attributes.

Reply:
checking deeply, an obsolete node sets its state to PersistWarning according to https://github.com/DynamoDS/Dynamo/blob/34aeb541af8cf5a370e76340f7bc5c398fdf7d8e/src/DynamoCore/Graph/Nodes/ZeroTouch/DSFunctionBase.cs#L265 and https://github.com/DynamoDS/Dynamo/blob/34aeb541af8cf5a370e76340f7bc5c398fdf7d8e/src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs#L1476 , so implicitly is already handling by the HasWarnings property.